c7n/resources/rdscluster.py

```diff
@@ -476,6 +478,61 @@ class RDSSnapshotAge(AgeFilter):
     date_attribute = 'SnapshotCreateTime'
 
 
+@RDSClusterSnapshot.action_registry.register('set-permissions')
+class SetPermissions(rds.SetPermissions):
+    """Set permissions for copying or restoring an RDS cluster snapshot
+
+    Use the 'add' and 'remove' parameters to control which accounts to
+    add or remove, respectively.  The default is to remove any
+    permissions granted to other AWS accounts.
+
+    Use `remove: matched` in combination with the `cross-account` filter
+    for more flexible removal options such as preserving access for
+    a set of whitelisted accounts:
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rds-cluster-snapshot-prune-permissions
+                resource: rds-cluster-snapshot
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - '112233445566'
+                actions:
+                  - type: set-permissions
+                    remove: matched
+    """
+    permissions = ('rds:ModifyDBClusterSnapshotAttribute',)
+
+    def process_snapshot(self, client, snapshot):
+        add_accounts = self.data.get('add', [])
+        remove_accounts = self.data.get('remove', [])
+
+        if not (add_accounts or remove_accounts):
+            if CrossAccountSnapshot.attributes_key not in snapshot:
+                attrs = {
+                    t['AttributeName']: t['AttributeValues']
+                    for t in self.manager.retry(
+                        client.describe_db_cluster_snapshot_attributes,
+                        DBClusterSnapshotIdentifier=snapshot['DBClusterSnapshotIdentifier']
+                    )['DBClusterSnapshotAttributesResult']['DBClusterSnapshotAttributes']
+                }
+                snapshot[CrossAccountSnapshot.attributes_key] = attrs
+            remove_accounts = snapshot[CrossAccountSnapshot.attributes_key].get('restore', [])
+        elif remove_accounts == 'matched':
+            remove_accounts = snapshot.get(CrossAccountSnapshot.annotation_key, [])
+
+        if add_accounts or remove_accounts:
+            client.modify_db_cluster_snapshot_attribute(
```

Comment:
this should ideally not be passing parameters that aren't set, wrt to add_accounts or remove_accounts being empty values.

Reply:
from the tests it looks like these don't cause any issues with the api when giving empty values?

Reply:
yeah, the api is fine taking empty lists for those values. i do see your point though about not passing the param at all, if it wasn't set in the policy and would be empty anyway.